zcash_client_backend/src/fees/common.rs

```diff
@@ -91,38 +103,40 @@ where
     })
 }
 
-pub(crate) fn single_change_output_policy<NoteRefT: Clone, F: FeeRule, E>(
-    _net_flows: &NetFlows,
-    _fallback_change_pool: ShieldedProtocol,
-) -> Result<(ShieldedProtocol, usize, usize), ChangeError<E, NoteRefT>>
-where
-    E: From<F::Error> + From<BalanceError>,
-{
+#[allow(clippy::bool_to_int_with_if)]
+pub(crate) fn single_change_output_policy(
+    net_flows: &NetFlows,
+    fallback_change_pool: ShieldedProtocol,
+) -> Result<(ShieldedProtocol, usize, usize), BalanceError> {
     // TODO: implement a less naive strategy for selecting the pool to which change will be sent.
+
+    // If the flows are not transparent, there is always change (possibly zero-valued).
```

Comment:
The comment was not correct for the old code. The reason for it being incorrect for the old code is not in this function; it's due to the PR's modifications to `single_change_output_policy`. Previously, `single_change_output_policy` had:

```rust
if proposed_change.is_zero()
```
where it now has
```rust
if proposed_change.is_zero() && sapling_change == 0 && orchard_change == 0
```
In other words, `sapling_change` and `orchard_change` were ignored when it came to constructing the `TransactionBalance`. But not completely ignored, because `orchard_change` was still used to calculate the number of outputs passed to the bundle's `num_actions()` method. That was a bug, I believe.

Reply:
The comment above no longer applies.